File: CS/WebSample/WebSample.Module.Web/FilterRowController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Web.Editors.ASPx;
using WinSample.Module;
using DevExpress.ExpressApp.NodeWrappers;
using System.Net;
using DevExpress.ExpressApp.Web.SystemModule;
using DevExpress.Web.ASPxGridView;

namespace WebSample.Module.Web {
    public partial class FilterRowController : ViewController {
        public FilterRowController() {
            InitializeComponent();
            RegisterActions(components);
            TargetViewId = "Person_ListView";
        }
        protected override void OnActivated() {
            base.OnActivated();
            IModelListViewWeb modelListViewWeb = ((IModelListViewWeb)View.Model);
            if (modelListViewWeb.FilterExpression == string.Empty)
            {
                modelListViewWeb.FilterExpression = "[Country] = 'USA'";
            }

        }
        private void changeFilterAction_Execute(object sender, SimpleActionExecuteEventArgs e) {
            if (View.IsControlCreated) {
                ASPxGridView control = (View as ListView).Editor.Control as ASPxGridView;
                control.FilterExpression = "[Country] = 'UK'";
            }
        }
    }
}

File: VB/VB/WebSample.Module.Web/FilterRowController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Web.Editors.ASPx
Imports WinSample.Module
Imports DevExpress.ExpressApp.NodeWrappers
Imports System.Net
Imports DevExpress.ExpressApp.Web.SystemModule
Imports DevExpress.Web.ASPxGridView

Namespace WebSample.Module.Web
	Partial Public Class FilterRowController
		Inherits ViewController
		Public Sub New()
			InitializeComponent()
			RegisterActions(components)
			TargetViewId = "Person_ListView"
		End Sub
		Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            Dim modelListViewWeb As IModelListViewWeb = TryCast(View.Model, IModelListViewWeb)
            If modelListViewWeb.FilterExpression = String.Empty Then
                modelListViewWeb.FilterExpression = "[Country] = 'USA'"
            End If
        End Sub
		Private Sub changeFilterAction_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs) Handles changeFilterAction.Execute
			If View.IsControlCreated Then
                Dim control As ASPxGridView = TryCast((TryCast(View, ListView)).Editor.Control, ASPxGridView)
				control.FilterExpression = "[Country] = 'UK'"
			End If
		End Sub
	End Class
End Namespace
